### ISSUE TITLE
Issues with postgres 

### ISSUE THREAD

**Issue by vacquah** [TobikoData/sqlmesh#3067]
I am having issues running pip install sqlmesh[postgres] on an Ubuntu 22.04 machine. The errors are related to the install of psycopg. I installed psycopg2-binary manually and tried again. Got the error below:

```
nv/include -I/usr/include/python3.10 -I. -I/usr/include/postgresql -I/usr/include/postgresql/14/server -I/usr/include/libxml2 -c psycopg/adapter_asis.c -o build/temp.linux-x86_64-3.10/psycopg/adapter_asis.o -Wdeclaration-after-statement
      In file included from psycopg/adapter_asis.c:28:
      ./psycopg/psycopg.h:35:10: fatal error: Python.h: No such file or directory
         35 | #include <Python.h>
            |          ^~~~~~~~~~
      compilation terminated.
      
      It appears you are missing some prerequisite to build the package from source.
      
      You may install a binary package by installing 'psycopg2-binary' from PyPI.
      If you want to install psycopg2 from source, please install the packages
      required for the build and try again.
      
      For further information please check the 'doc/src/install.rst' file (also at
      <https://www.psycopg.org/docs/install.html>).
      
      error: command '/usr/bin/x86_64-linux-gnu-gcc' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> psycopg2

note: This is an issue with the package mentioned above, not pip.
hint: See above for output from the failure.
```

**Comment by vvcb** [comment]
@georgesittas , should this issue be reopened?

`pip install sqlmesh[postgres]` or `uv add sqlmesh[postgres]` result in the same error that @vacquah reported, both on Windows 11 and Ubuntu. Installing `libpq-dev` using apt resolved this on Ubuntu. Python=3.12.8

I can see that `psycopg2-binary` is listed as a [dev dependency](https://github.com/TobikoData/sqlmesh/blob/774cff7dcff2772d17adfe27c8a0c4838d82d389/setup.py#L83). Could/Should the same be done for the [optional dependency](https://github.com/TobikoData/sqlmesh/blob/774cff7dcff2772d17adfe27c8a0c4838d82d389/setup.py#L138) as well? 

```bash
jovyan@61737ab111f6:~/test$ uv add sqlmesh[postgres]
Resolved 167 packages in 1.70s
  × Failed to build `psycopg2==2.9.10`
  ├─▶ The build backend returned an error
  ╰─▶ Call to `setuptools.build_meta:__legacy__.build_wheel` failed (exit status: 1)

      [stdout]
      running egg_info
      writing psycopg2.egg-info/PKG-INFO
      writing dependency_links to psycopg2.egg-info/dependency_links.txt
      writing top-level names to psycopg2.egg-info/top_level.txt

      [stderr]

      Error: pg_config executable not found.

      pg_config is required to build psycopg2 from source.  Please add the directory
      containing pg_config to the $PATH or specify the full executable path with the
      option:

          python setup.py build_ext --pg-config /path/to/pg_config build ...

      or with the pg_config option in 'setup.cfg'.

      If you prefer to avoid building psycopg2 from source, please install the PyPI
      'psycopg2-binary' package instead.

      For further information please check the 'doc/src/install.rst' file (also at
      <https://www.psycopg.org/docs/install.html>).


      hint: This usually indicates a problem with the package or the build environment.
```

**Comment by Haklim733** [comment]
`apt install clang` resolved the issue for me. However, I think a dependency on psycopg2-binary rather than psycopg2 would be an improvement if possible.
